torch/_inductor/codegen/cpp_prefix.h

```diff
@@ -62,6 +62,13 @@ uint64_t randint64_cpu(uint32_t seed, uint32_t offset, int64_t low, int64_t high
   return (result % static_cast<uint64_t>(high - low)) + low;
 }
 
+template <typename T> void atomic_add(volatile T *addr, T offset) {
+  static_assert(sizeof(std::atomic<T>) == sizeof(T),
+                "std::atomic issue");
+  std::atomic<T> *atomic_addr = (std::atomic<T> *)addr;
+  atomic_addr->fetch_add(offset, std::memory_order_relaxed);
+}
+
```

Comment:
I would suggest we do `fetch_add` with integral types first considering you proved the perf gains and the support is available since c++11.